### PR TITLE
Additional options native embeddings

### PR DESCRIPTION
**Pull request by BernhardEisvogel**
Hey everyone!

I want to use this repository in a language program that need the word embedding instead of the sentence embedding, so I changed a way to switch between those two modes in NativeEmbeddings.init.

Greetings Bernhard
P.s. I'm new to Github

### CODE REVIEW

File: Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift [hunk 1]
```diff
@@ -8,26 +8,40 @@
...
+    case sentenceEmbedding
+}
+
 @available(macOS 11.0, iOS 15.0, *)
 public class NativeEmbeddings: EmbeddingsProtocol {
     public let model: ModelActor
     public let tokenizer: any TokenizerProtocol
 
-    public init(language: NLLanguage = .english) {
+    public init(language: NLLanguage = .english, type:NativeEmbeddingType = .sentenceEmbedding) {
```
Comment (ZachNagengast) [inline Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift:21]:
```suggestion
    public init(language: NLLanguage = .english, type: NativeEmbeddingType = .sentenceEmbedding) {
```

File: Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift [hunk 2]
```diff
@@ -8,26 +8,40 @@
...
-                fatalError("Failed to load the Core ML model.")
+        init(language: NLLanguage, type:NativeEmbeddingType = .sentenceEmbedding) {
+            switch type {
+                case .sentenceEmbedding:
+                    guard let nativeModel = NLEmbedding.sentenceEmbedding(for: language) else {
+                        fatalError("Failed to load the Core ML model.")
+                    }
+                    model = nativeModel
+                case.wordEmbedding:
+                    guard let nativeModel = NLEmbedding.sentenceEmbedding(for: language) else {
```
Comment (ZachNagengast) [inline Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift:39]:
```suggestion
                case .wordEmbedding:
                    guard let nativeModel = NLEmbedding.wordEmbedding(for: language) else {
```

Is this the model you wanted to use?

File: Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift [hunk 3]
```diff
@@ -8,26 +8,40 @@
...
 
     // MARK: - Dense Embeddings
 
     public actor ModelActor {
         private let model: NLEmbedding
 
-        init(language: NLLanguage) {
-            guard let nativeModel = NLEmbedding.sentenceEmbedding(for: language) else {
-                fatalError("Failed to load the Core ML model.")
+        init(language: NLLanguage, type:NativeEmbeddingType = .sentenceEmbedding) {
```
Comment (ZachNagengast) [inline Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeEmbeddings.swift:31]:
```suggestion
        init(language: NLLanguage, type: NativeEmbeddingType = .sentenceEmbedding) {
```

**Review by ZachNagengast — CHANGES_REQUESTED** [review]
Hey @BernhardEisvogel, great idea! I just made couple suggestions for edits, otherwise looks great!

**Comment by BernhardEisvogel** [comment]
Great! I will upload another commit later improving the performance of the search algorithm, currently it's ordering the complete list and then it takes "the best" k entries( in  O(nlogn) )but it's sufficient to use some topK algorithm ( in O(n) )

**Review by ZachNagengast — APPROVED** [review]
Looks good! Thanks @BernhardEisvogel

**Pull request merged**